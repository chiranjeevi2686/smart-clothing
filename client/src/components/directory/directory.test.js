import React from "react";
import { render, fireEvent } from '@testing-library/react';
import Directory from './directory.component';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const defaultStore =
{
    directory: {
        sections: [{
            id: 1,
            imageUrl: "https://i.ibb.co/cvpntL1/hats.png",
            linkUrl: "shop/hats",
            size: "",
            title: "hats"
        }]
    }
};
const mockStore = configureStore();
describe('Directory component', () => {
    it('should render directory', async () => {
        const { asFragment, getByTestId, getByText } = render(<Provider store={mockStore(defaultStore)}>
            <Directory />
        </Provider>)
        expect(asFragment()).toMatchSnapshot();
        // const displayNameInput = getByTestId('display-name');
        // fireEvent.change(displayNameInput, {target: {value:'testDisplay'}})
        // expect(displayNameInput.value).toEqual('testDisplay')

    })
})
