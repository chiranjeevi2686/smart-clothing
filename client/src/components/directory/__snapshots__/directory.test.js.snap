// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory component should render directory 1`] = `
<DocumentFragment>
  <div
    class="sc-jSgupP gFhuWt"
  >
    <div
      class="sc-bdfBwQ cueKct"
    >
      <div
        class="sc-gsTCUz CKdSB background-image"
      />
      <div
        class="sc-dlfnbm JasYd"
      >
        <span
          class="sc-hKgILt eLkpOW"
        >
          HATS
        </span>
        <span
          class="sc-eCssSg cAcTCL"
        >
          SHOP NOW
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
